test/rest-api/response-patching.test.ts

```diff
@@ -161,34 +157,7 @@ describe('REST: Response patching', () => {
 
       expect(status).toBe(200)
       expect(headers).toHaveProperty('x-powered-by', 'msw')
-      expect(body).toEqual(null)
-    })
-  })
-
-  describe('given a HEAD request to be patched', () => {
-    it('should be able to properly request and patch the response', async () => {
-      const res = await test.request({
-        url: '/posts',
-        fetchOptions: {
-          method: 'HEAD',
-          headers: {
-            'Content-Type': 'application/json',
-          },
-        },
-        responsePredicate(res, url) {
-          return (
-            match(test.origin + url, res.url()).matches &&
-            res.headers()['x-powered-by'] === 'msw'
-          )
-        },
-      })
-      const status = res.status()
-      const headers = res.headers()
-      const body = await res.json()
-
-      expect(status).toBe(200)
-      expect(headers).toHaveProperty('x-powered-by', 'msw')
-      expect(body).toEqual(null)
```

Comment:
we can keep the test with HEAD method

Reply:
I tried to create a test for the `HEAD` method. Unfortunately I always get a `TimeoutError: Timeout exceeded while waiting for event`. Am I doing something wrong here? Would appreciate some help 😅

Reply:
It looks like the `responsePredicate` is returning `false`, which is causing the `TimeoutError` you see. If you just `return true` there, you'll run into: 

```
  ● REST: Response patching › given a HEAD request to be patched › should be able to properly request and patch the response

    Protocol error (Network.getResponseBody): No data found for resource with given identifier
```

You'll want to make sure that you add handlers for the route. 

Try this out. in `response-patching.mocks.ts`, add:
```js
  rest.head('/headtest', async (req, res, ctx) => {
    return res(
      ctx.json(null),
    )
  }),
```

In `response-patching.test.ts`:

```js
withRoutes(app) {
          app.post('/headers-proxy', (req, res) => {
            const { authorization } = req.headers

            if (!authorization) {
              return res.status(403).json({ message: 'error' }).end()
            }

            return res.status(200).json({ message: 'success' }).end()
          })

          app.post('/posts', (req, res) => {
            res.status(200).json({ id: 101 }).end()
          })

          app.head('/posts', (req, res) => {
            res.status(200).end()
          })

          app.get('/posts', (req, res) => {
            res.status(200).json({ id: 101 }).end()
          })

          app.head('/headtest', (req, res) => {
            res.status(200).end()
          })
        },
```

And add the test:

```js
  describe('given a HEAD request to be patched', () => {
    it('should be able to properly request and patch the response', async () => {
      const res = await test.request({
        url: '/headtest',
        fetchOptions: {
          method: 'HEAD'
        },
        responsePredicate(res, url) {
          return (
            match(test.origin + url, res.url()).matches &&
            res.headers()['x-powered-by'] === 'msw'
          )
        },
      })
      const status = res.status()
      const headers = res.headers()
      const body = await res.json()

      expect(status).toBe(200)
      expect(headers).toHaveProperty('x-powered-by', 'msw')
      expect(body).toEqual(null)
    })
  })
```

Reply:
disclaimer: even though this works, @marcosvega91 or @kettanaito probably have a better recommendation here.

Reply:
@msutkowski Thank you for your help! Added the test. Lets wait for the feedback from @kettanaito or @marcosvega91  😀

Reply:
HEAD request should make the original request using `ctx.fetch` to make the test valid. The problem is that the head request is not resolving throwing the timeout error. @kettanaito have you any ideas of what the problem is ?

Reply:
I have found the problem. After the original fetch is called, a function `gracefully` will return the `json` or the `text` depending on the `header`. The problem is that when using method `HEAD` the header is set but the body will be always null. I don't know if is really correct to use `gracefully` function because then you cannot access headers in your mocked function 🤔